Mlem/Models/Content/Post/Post2Providing.swift

```diff
@@ -40,4 +42,74 @@ extension Post2Providing {
     var myVote_: ScoringOperation? { post2.myVote }
     
     var score: Int { upvoteCount - downvoteCount }
+    
+    func vote(_ newVote: ScoringOperation) async throws {
+        if newVote == myVote { return }
+        let oldVote = myVote
+        switch oldVote {
+        case .upvote:
+            upvoteCount -= 1
+        case .downvote:
+            downvoteCount -= 1
+        default:
+            break
+        }
+        switch newVote {
+        case .upvote:
+            upvoteCount += 1
+        case .downvote:
+            downvoteCount += 1
+        default:
+            break
+        }
+        myVote = newVote
```

Comment:
This seems like logic we're going to want to reuse--maybe put it in a `ScoringOperation` extension:
```
func computeVoteDelta(newVote) -> (Int, Int) {
  if newVote == self { return }
  var upvoteDelta = 0
  var downvoteDelta = 0
  switch self { ... }
  switch newVote { ... }

  return (upvoteDelta, downvoteDelta)
}

```

Reply:
Good idea. I've implemented something similar to this 👍